todo_app/app.py

```diff
@@ -21,10 +52,19 @@ def index():
         else:
             items.sort(key=lambda x: x.due_date)
 
-        return render_template('index.html', view_model=item_view_model, todays_date=todays_date)
+
+        if hasattr(current_user, 'role'):
+            if current_user.role == 'writer':
+                return render_template('index.html', view_model=item_view_model, todays_date=todays_date)   
+            else:
+                return render_template('index_ro.html', view_model=item_view_model, todays_date=todays_date)
+        else:
+            return render_template('index.html', view_model=item_view_model, todays_date=todays_date)
```

Comment:
Instead of using hasattr, you could check if login is disabled. There should be an error if login is not disabled but the user doesn't have a role.